cs/libdpr/samples/DprCounters/DprCounters/python/start_cluster.py

```diff
@@ -0,0 +1,250 @@
+from ctypes import ArgumentError
+from http import server
+from posixpath import basename
+import resource
+from kubernetes import client, config
+import os
+import time
+from numpy import void
+
+import yaml
+
+
+class KubernetesCluster():
+
+    DEFAULT_STORAGE = "1Gi"
+    SUPPORTED_SERVER_TYPES = ["counter"]
+    BASE_PORT = 6380
+    BASE_AZURE_PORT = 31101
+    LOCAL_DIRECTORY = "/mnt/c/Users/cetko/OneDrive/Desktop/MEng Thesis/new_faster/cs/libdpr/samples/DprCounters/DprCounters"
+    LOCAL_IMAGE = "cetko24/meng_project"
+    LOCAL_POLICY = "Never"
+    AZURE_DIRECTORY = "/home/nikola/FASTER/cs/libdpr/samples/DprCounters/DprCounters"
+    AZURE_IMAGE = "nikolameng.azurecr.io/meng"
+    AZURE_POLICY = "Always"
+
+
+    class Server():
+
+        def __init__(self, type:str, storage_location:str = None, storage_capacity:str = None, cpu_request:str = None, cpu_limit:str = None, memory_request:str = None, memory_limit:str = None):
+            self.type = type
+            self.storage_location = storage_location
+            self.storage_capacity = storage_capacity
+            self.cpu_request = cpu_request
+            self.cpu_limit = cpu_limit
+            self.memory_request = memory_request
+            self.memory_limit = memory_limit
+
+    def __init__(self, checkpoint_dir:str = None, azure:bool = False) -> void:
```

Comment:
Let's try to have a bit more sophistication around Kubernetes backend --- as opposed to have boolean flags around azure, can we have some kind of polymorphism around k8s backend? (e.g., minikubes & azure, and the ability to easily add others like Amazon without needing to change code.)